### DISCUSSION TITLE
SpirV code with full glsl source for debugging?

### DISCUSSION THREAD

**Discussion by Melchiah** [DiligentGraphics/DiligentEngine#265] [Q&A] [answered]
Hello, is it possible to load spirv directly, or atleast pass glslangValidator flags for keeping debug source of glsl (paramerter -gVS). This enables full glsl debuging in RenderDoc, which would be really helpful. Iam aware that you are patching the shader code and injecting info about descriptor sets, so iam not sure whether glsl source would still be valid (if you are patching it on that layer, then maybe). Iam using exlusivelly glsl. Thanks for answer!

**Comment by TheMostDiligent** [comment, marked as answer]
>  is it possible to load spirv directly

Yes, when you create the shader, you can provide the [bytecode directly](https://github.com/DiligentGraphics/DiligentCore/blob/2ebd6cd1b3221a50d7f11f77ac51023a2b33697f/Graphics/GraphicsEngine/interface/Shader.h#L369).

There is currently no way to pass command-line options, Diligent does not use the command line interface for shader compilation
